kolibri/plugins/learn/assets/src/my_downloads/views/DownloadsList/index.vue

```diff
@@ -128,25 +131,70 @@
     setup() {
       const { genExternalContentURLBackLinkCurrentPage } = useContentLink();
       const { getLearningActivityIcon } = useLearningActivities();
-
+      const { downloadRequestMap, availableSpace } = useDownloadRequests();
+      const totalPageNumber = ref(0);
+      const store = getCurrentInstance().proxy.$store;
+      const query = computed(() => get(route).query);
+      const route = computed(() => store.state.route);
+      const sort = computed(() => query.value.sort);
+      const pageSizeNumber = computed(() => Number(query.value.page_size || 25));
+      const activityType = computed(() => query.value.activity || 'all');
+      const downloads = ref([]);
+      const sortedFilteredDownloads = () => {
+        let downloadsToDisplay = [];
+        if (downloadRequestMap) {
+          for (const [, value] of Object.entries(downloadRequestMap.value)) {
+            downloadsToDisplay.push(value);
+          }
+          if (sort) {
+            switch (sort.value) {
+              case 'newest':
+                downloadsToDisplay.sort(
+                  (a, b) => new Date(b.requested_at) - new Date(a.requested_at)
+                );
+                break;
+              case 'oldest':
+                downloadsToDisplay.sort(
+                  (a, b) => new Date(a.requested_at) - new Date(b.requested_at)
+                );
+                break;
+              case 'smallest':
+                downloadsToDisplay.sort((a, b) => a.metadata.file_size - b.metadata.file_size);
+                break;
+              case 'largest':
+                downloadsToDisplay.sort((a, b) => b.metadata.file_size - a.metadata.file_size);
+                break;
+              default:
+                // If no valid sort option provided, return unsorted array
+                break;
+            }
+          }
+          if (activityType) {
```

Comment:
If we filter by activityType first, then potentially, we won't have to sort as many things (not a huge deal).